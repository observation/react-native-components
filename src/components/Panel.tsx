import React from 'react'
import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native'

import theme from '../styles/theme'

type Props = {
  children?: React.ReactNode
  containerStyle?: StyleProp<ViewStyle>
  panelStyle?: StyleProp<ViewStyle>
  topShadow?: boolean
}

/** Display content in a full width panel with a shadow */
const Panel = ({ children, topShadow = true, containerStyle, panelStyle }: Props) => {
  return (
    <View style={[styles.panelContainer, topShadow ? {} : { marginTop: 0 }, containerStyle]}>
      <View style={[styles.panel, topShadow ? {} : { paddingTop: 0, borderTopWidth: 0 }, panelStyle]}>{children}</View>
    </View>
  )
}

export default Panel

const styles = StyleSheet.create({
  panelContainer: {
    marginTop: theme.margin.quarter,
    ...theme.shadowSmall.ios,
  },
  panel: {
    paddingVertical: theme.margin.common,
    backgroundColor: theme.color.white,
    ...theme.shadowSmall.android,
    borderTopWidth: 1,
    borderColor: theme.color.greyLight,
  },
})
