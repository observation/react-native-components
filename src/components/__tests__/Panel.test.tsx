import React from 'react'
import { Text } from 'react-native'

import { render } from '@testing-library/react-native'

import Panel from '../Panel'

describe('Panel', () => {
  test('Rendering', () => {
    const { toJSON } = render(
      <Panel>
        <Text>Some text</Text>
      </Panel>,
    )
    expect(toJSON()).toMatchSnapshot()
  })

  test('Rendering without top shadow', () => {
    const { toJSON } = render(
      <Panel topShadow={false}>
        <Text>Some text</Text>
      </Panel>,
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
