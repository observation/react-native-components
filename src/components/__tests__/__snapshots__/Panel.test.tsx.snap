// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel Rendering 1`] = `
<View
  style={
    [
      {
        "marginTop": 4,
        "shadowColor": "#E6E6E6",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.8,
        "shadowRadius": 2,
      },
      {},
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#F8F9FA",
          "borderTopWidth": 1,
          "elevation": 4,
          "paddingVertical": 16,
          "shadowColor": "#939393",
        },
        {},
        undefined,
      ]
    }
  >
    <Text>
      Some text
    </Text>
  </View>
</View>
`;

exports[`Panel Rendering without top shadow 1`] = `
<View
  style={
    [
      {
        "marginTop": 4,
        "shadowColor": "#E6E6E6",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.8,
        "shadowRadius": 2,
      },
      {
        "marginTop": 0,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#F8F9FA",
          "borderTopWidth": 1,
          "elevation": 4,
          "paddingVertical": 16,
          "shadowColor": "#939393",
        },
        {
          "borderTopWidth": 0,
          "paddingTop": 0,
        },
        undefined,
      ]
    }
  >
    <Text>
      Some text
    </Text>
  </View>
</View>
`;
